Client/AdjustHelper.swift

```diff
@@ -0,0 +1,94 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Adjust
+import Shared
+
+private let log = Logger.browserLogger
+
+final class AdjustHelper {
+
+    private static let adjustAppTokenKey = "AdjustAppToken"
+    private let profile: Profile
+
+    init(profile: Profile) {
+        self.profile = profile
+    }
+
+    func setupAdjust() {
+        guard let config = getConfig() else { return }
+
+        // Always initialize Adjust if we have a config - otherwise we cannot enable/disable it later. Their SDK must be
+        // initialized through appDidFinishLaunching otherwise it will be in a bad state.
+        Adjust.appDidLaunch(config)
+
+        AdjustHelper.setEnabled(shouldEnable)
+    }
+
+    /// Used to enable or disable Adjust SDK and it's features. We disable third party sharing by default.
+    /// If user has disabled Send Anonymous Usage Data then we ask Adjust to erase the user's data as well.
+    static func setEnabled(_ enabled: Bool) {
+        Adjust.disableThirdPartySharing()
```

Comment:
Q. Considering we are always disabling first. Have we asked or looked at their api that it's okay to disable it first?

ref: https://github.com/adjust/android_sdk/issues/408 - This is an older issue but its good to document what we do here.

Reply:
Yes, it was determined it was ok for us to disable by default third party sharing after receiving the following answer from Adjust:

> The disableThirdPartySharing functionality is typically used for apps that give users the option to opt out of Third 
> Party Sharing, and is usually implemented on the user level. If this is implemented for all users, it limits the ability 
> to share attribution data with partners (such as Facebook and Google) and will limit their ability to receive 
> callbacks from Adjust, directly affecting the attribution ability of partners.

Since this use case doesn't apply to us, it was decided to disable by default.